surveyor/jetstream_test.go

```diff
@@ -0,0 +1,137 @@
+package surveyor
+
+import (
+	"bytes"
+	"testing"
+	"time"
+
+	"github.com/nats-io/jsm.go"
+	"github.com/nats-io/nats.go"
+	ptu "github.com/prometheus/client_golang/prometheus/testutil"
+
+	st "github.com/nats-io/nats-surveyor/test"
+)
+
+func TestJetStream_Load(t *testing.T) {
+	js := st.NewJetStreamServer(t)
+	defer js.Shutdown()
+
+	opt := GetDefaultOptions()
+	opt.URLs = js.ClientURL()
+
+	obs, err := NewJetStreamAdvisoryListener("testdata/goodjs/global.json", *opt)
+	if err != nil {
+		t.Fatalf("jetstream load error: %s", err)
+	}
+	obs.Stop()
+
+	_, err = NewJetStreamAdvisoryListener("testdata/badjs/missing.json", *opt)
+	if err.Error() != "open testdata/badjs/missing.json: no such file or directory" {
+		t.Fatalf("jetstream load error: %s", err)
+	}
+
+	_, err = NewJetStreamAdvisoryListener("testdata/badobs/bad.json", *opt)
+	if err.Error() != "invalid JetStream advisory configuration: testdata/badobs/bad.json: name is required" {
+		t.Fatalf("jetstream load error: %s", err)
+	}
+}
+
+func TestJetStream_Handle(t *testing.T) {
+	js := st.NewJetStreamServer(t)
+	defer js.Shutdown()
+
+	opt := GetDefaultOptions()
+	opt.URLs = js.ClientURL()
+
+	obs, err := NewJetStreamAdvisoryListener("testdata/goodjs/global.json", *opt)
+	if err != nil {
+		t.Fatalf("jetstream load error: %s", err)
+	}
+	defer obs.Stop()
+
+	err = obs.Start()
+	if err != nil {
+		t.Fatalf("jetstream failed to start: %s", err)
+	}
+
```

Comment:
Nitpick...  should probably move the `defer obs.Stop()` above to right here.

Reply:
ah, not a Nipick, its just wrong :)